### ISSUE TITLE
error on distributed inference

### ISSUE THREAD

**Issue by vigneshgig** [georgian-io/Knowledge-Distillation-Toolkit#1]
torch version:'1.8.1+cu102'
python: 3.7.5
ubuntu: 16.04
ray:'1.3.0'

 0%|          | 0/2703 [00:00<?, ?it/s]
2021-05-24 14:18:24,104	WARNING worker.py:1115 -- Failed to unpickle the remote function '__main__.remote_process_data_sample' with function ID 7e54bbdac7183014626c026a1a5f7a5d94bd047c2c579b3d7b5d492e. Traceback:
Traceback (most recent call last):
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/ray/_private/function_manager.py", line 180, in fetch_and_register_remote_function
    function = pickle.loads(serialized_function)
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/ray/cloudpickle/cloudpickle.py", line 584, in subimport
    __import__(name)
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/torch/__init__.py", line 635, in <module>
    import torch.utils.data
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/torch/utils/data/__init__.py", line 6, in <module>
    from .dataloader import DataLoader, _DatasetKind, get_worker_info
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/torch/utils/data/dataloader.py", line 21, in <module>
    from . import _utils
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/torch/utils/data/_utils/__init__.py", line 45, in <module>
    from . import worker, signal_handling, pin_memory, collate, fetch
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/torch/utils/data/_utils/worker.py", line 114, in <module>
    class _IterableDatasetStopIteration(object):
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/dataclasses.py", line 950, in wrap
    return _process_class(cls, init, repr, eq, order, unsafe_hash, frozen)
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/dataclasses.py", line 801, in _process_class
    for name, type in cls_annotations.items()]
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/dataclasses.py", line 801, in <listcomp>
    for name, type in cls_annotations.items()]
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/dataclasses.py", line 659, in _get_field
    if (_is_classvar(a_type, typing)
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/dataclasses.py", line 550, in _is_classvar
    return type(a_type) is typing._ClassVar
AttributeError: module 'typing' has no attribute '_ClassVar'

2021-05-24 14:18:24,106	WARNING worker.py:1115 -- Failed to unpickle the remote function '__main__.remote_process_data_sample' with function ID 7e54bbdac7183014626c026a1a5f7a5d94bd047c2c579b3d7b5d492e. Traceback:
Traceback (most recent call last):
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/ray/_private/function_manager.py", line 180, in fetch_and_register_remote_function
    function = pickle.loads(serialized_function)
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/ray/cloudpickle/cloudpickle.py", line 584, in subimport
    __import__(name)
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/torch/__init__.py", line 635, in <module>
    import torch.utils.data
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/torch/utils/data/__init__.py", line 6, in <module>
    from .dataloader import DataLoader, _DatasetKind, get_worker_info
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/torch/utils/data/dataloader.py", line 21, in <module>
    from . import _utils
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/torch/utils/data/_utils/__init__.py", line 45, in <module>
    from . import worker, signal_handling, pin_memory, collate, fetch
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/torch/utils/data/_utils/worker.py", line 114, in <module>
    class _IterableDatasetStopIteration(object):
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/dataclasses.py", line 950, in wrap
    return _process_class(cls, init, repr, eq, order, unsafe_hash, frozen)
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/dataclasses.py", line 801, in _process_class
    for name, type in cls_annotations.items()]
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/dataclasses.py", line 801, in <listcomp>
    for name, type in cls_annotations.items()]
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/dataclasses.py", line 659, in _get_field
    if (_is_classvar(a_type, typing)
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/dataclasses.py", line 550, in _is_classvar
    return type(a_type) is typing._ClassVar
AttributeError: module 'typing' has no attribute '_ClassVar'

(pid=5512) 2021-05-24 14:18:24,096	ERROR serialization.py:248 -- name '_C' is not defined
(pid=5512) Traceback (most recent call last):
(pid=5512)   File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/ray/serialization.py", line 246, in deserialize_objects
(pid=5512)     obj = self._deserialize_object(data, metadata, object_ref)
(pid=5512)   File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/ray/serialization.py", line 188, in _deserialize_object
(pid=5512)     return self._deserialize_msgpack_data(data, metadata_fields)
(pid=5512)   File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/ray/serialization.py", line 166, in _deserialize_msgpack_data
(pid=5512)     python_objects = self._deserialize_pickle5_data(pickle5_data)
(pid=5512)   File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/ray/serialization.py", line 156, in _deserialize_pickle5_data
(pid=5512)     obj = pickle.loads(in_band)
(pid=5512)   File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/torch/__init__.py", line 228, in <module>
(pid=5512)     __all__ += [name for name in dir(_C)
(pid=5512) NameError: name '_C' is not defined
(pid=5512) 2021-05-24 14:18:24,096	ERROR serialization.py:248 -- name '_C' is not defined
(pid=5512) Traceback (most recent call last):
(pid=5512)   File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/ray/serialization.py", line 246, in deserialize_objects
(pid=5512)     obj = self._deserialize_object(data, metadata, object_ref)
(pid=5512)   File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/ray/serialization.py", line 188, in _deserialize_object
(pid=5512)     return self._deserialize_msgpack_data(data, metadata_fields)
(pid=5512)   File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/ray/serialization.py", line 166, in _deserialize_msgpack_data
(pid=5512)     python_objects = self._deserialize_pickle5_data(pickle5_data)
(pid=5512)   File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/ray/serialization.py", line 156, in _deserialize_pickle5_data
(pid=5512)     obj = pickle.loads(in_band)
(pid=5512)   File "/mnt/media/Knowledge-Distillation-Toolkit/blog_post_resources/wav2vec2_inference_demo/utils.py", line 3, in <module>
(pid=5512)     import torch
(pid=5512)   File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/torch/__init__.py", line 228, in <module>
(pid=5512)     __all__ += [name for name in dir(_C)
(pid=5512) NameError: name '_C' is not defined
(pid=5512) 2021-05-24 14:18:24,097	ERROR serialization.py:248 -- name '_C' is not defined
(pid=5512) Traceback (most recent call last):
(pid=5512)   File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/ray/serialization.py", line 246, in deserialize_objects
(pid=5512)     obj = self._deserialize_object(data, metadata, object_ref)
(pid=5512)   File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/ray/serialization.py", line 188, in _deserialize_object
(pid=5512)     return self._deserialize_msgpack_data(data, metadata_fields)
(pid=5512)   File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/ray/serialization.py", line 166, in _deserialize_msgpack_data
(pid=5512)     python_objects = self._deserialize_pickle5_data(pickle5_data)
(pid=5512)   File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/ray/serialization.py", line 156, in _deserialize_pickle5_data
(pid=5512)     obj = pickle.loads(in_band)
(pid=5512)   File "/mnt/media/Knowledge-Distillation-Toolkit/blog_post_resources/wav2vec2_inference_demo/utils.py", line 3, in <module>
(pid=5512)     import torch
(pid=5512)   File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/torch/__init__.py", line 228, in <module>
(pid=5512)     __all__ += [name for name in dir(_C)
(pid=5512) NameError: name '_C' is not defined
(pid=5512) 2021-05-24 14:18:24,098	ERROR serialization.py:248 -- name '_C' is not defined
(pid=5512) Traceback (most recent call last):
(pid=5512)   File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/ray/serialization.py", line 246, in deserialize_objects
(pid=5512)     obj = self._deserialize_object(data, metadata, object_ref)
(pid=5512)   File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/ray/serialization.py", line 188, in _deserialize_object
(pid=5512)     return self._deserialize_msgpack_data(data, metadata_fields)
(pid=5512)   File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/ray/serialization.py", line 166, in _deserialize_msgpack_data
(pid=5512)     python_objects = self._deserialize_pickle5_data(pickle5_data)
(pid=5512)   File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/ray/serialization.py", line 156, in _deserialize_pickle5_data2021-05-24 14:18:24,108	WARNING worker.py:1115 -- Failed to unpickle the remote function '__main__.remote_process_data_sample' with function ID 7e54bbdac7183014626c026a1a5f7a5d94bd047c2c579b3d7b5d492e. Traceback:
Traceback (most recent call last):
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/ray/_private/function_manager.py", line 180, in fetch_and_register_remote_function
    function = pickle.loads(serialized_function)
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/ray/cloudpickle/cloudpickle.py", line 584, in subimport
    __import__(name)
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/torch/__init__.py", line 635, in <module>
    import torch.utils.data
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/torch/utils/data/__init__.py", line 6, in <module>
    from .dataloader import DataLoader, _DatasetKind, get_worker_info
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/torch/utils/data/dataloader.py", line 21, in <module>
    from . import _utils
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/torch/utils/data/_utils/__init__.py", line 45, in <module>
    from . import worker, signal_handling, pin_memory, collate, fetch
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/torch/utils/data/_utils/worker.py", line 114, in <module>
    class _IterableDatasetStopIteration(object):
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/dataclasses.py", line 950, in wrap
    return _process_class(cls, init, repr, eq, order, unsafe_hash, frozen)
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/dataclasses.py", line 801, in _process_class
    for name, type in cls_annotations.items()]
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/dataclasses.py", line 801, in <listcomp>
    for name, type in cls_annotations.items()]
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/dataclasses.py", line 659, in _get_field
    if (_is_classvar(a_type, typing)
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/dataclasses.py", line 550, in _is_classvar
    return type(a_type) is typing._ClassVar
AttributeError: module 'typing' has no attribute '_ClassVar'

(pid=5512)     obj = pickle.loads(in_band)
(pid=5512)   File "/mnt/media/Knowledge-Distillation-Toolkit/utils/fairseq_mod/fairseq_mod/__init__.py", line 12, in <module>
(pid=5512)     from fairseq_mod.logging import meters, metrics, progress_bar  # noqa
(pid=5512)   File "/mnt/media/Knowledge-Distillation-Toolkit/utils/fairseq_mod/fairseq_mod/logging/meters.py", line 12, in <module>
(pid=5512)     import torch
(pid=5512)   File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/torch/__init__.py", line 228, in <module>
(pid=5512)     __all__ += [name for name in dir(_C)
(pid=5512) NameError: name '_C' is not defined
2021-05-24 14:18:24,111	WARNING worker.py:1115 -- Failed to unpickle the remote function '__main__.remote_process_data_sample' with function ID 7e54bbdac7183014626c026a1a5f7a5d94bd047c2c579b3d7b5d492e. Traceback:
Traceback (most recent call last):
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/ray/_private/function_manager.py", line 180, in fetch_and_register_remote_function
    function = pickle.loads(serialized_function)
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/ray/cloudpickle/cloudpickle.py", line 584, in subimport
    __import__(name)
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/torch/__init__.py", line 635, in <module>
    import torch.utils.data
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/torch/utils/data/__init__.py", line 6, in <module>
    from .dataloader import DataLoader, _DatasetKind, get_worker_info
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/torch/utils/data/dataloader.py", line 21, in <module>
    from . import _utils
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/torch/utils/data/_utils/__init__.py", line 45, in <module>
    from . import worker, signal_handling, pin_memory, collate, fetch
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/torch/utils/data/_utils/worker.py", line 114, in <module>
    class _IterableDatasetStopIteration(object):
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/dataclasses.py", line 950, in wrap
    return _process_class(cls, init, repr, eq, order, unsafe_hash, frozen)
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/dataclasses.py", line 801, in _process_class
    for name, type in cls_annotations.items()]
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/dataclasses.py", line 801, in <listcomp>
    for name, type in cls_annotations.items()]
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/dataclasses.py", line 659, in _get_field
    if (_is_classvar(a_type, typing)
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/dataclasses.py", line 550, in _is_classvar
    return type(a_type) is typing._ClassVar
AttributeError: module 'typing' has no attribute '_ClassVar'

---------------------------------------------------------------------------
RayTaskError                              Traceback (most recent call last)
<ipython-input-13-5050b0dbff6b> in <module>
      5     ground_truths.append(batch[2][0])
      6     break
----> 7 predictions = ray.get(prediction_futures)
      8 inference_time = time.time() - start_time

~/wav2vec_stt/lib/python3.7/site-packages/ray/_private/client_mode_hook.py in wrapper(*args, **kwargs)
     45         if client_mode_should_convert():
     46             return getattr(ray, func.__name__)(*args, **kwargs)
---> 47         return func(*args, **kwargs)
     48 
     49     return wrapper

~/wav2vec_stt/lib/python3.7/site-packages/ray/worker.py in get(object_refs, timeout)
   1479                     worker.core_worker.dump_object_store_memory_usage()
   1480                 if isinstance(value, RayTaskError):
-> 1481                     raise value.as_instanceof_cause()
   1482                 else:
   1483                     raise value

RayTaskError: ray::remote_process_data_sample() (pid=5512, ip=10.101.1.239)
  File "python/ray/_raylet.pyx", line 458, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 479, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 349, in ray._raylet.raise_if_dependency_failed
ray.exceptions.RaySystemError: System error: name '_C' is not defined
traceback: Traceback (most recent call last):
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/ray/serialization.py", line 246, in deserialize_objects
    obj = self._deserialize_object(data, metadata, object_ref)
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/ray/serialization.py", line 188, in _deserialize_object
    return self._deserialize_msgpack_data(data, metadata_fields)
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/ray/serialization.py", line 166, in _deserialize_msgpack_data
    python_objects = self._deserialize_pickle5_data(pickle5_data)
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/ray/serialization.py", line 156, in _deserialize_pickle5_data
    obj = pickle.loads(in_band)
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/torch/__init__.py", line 228, in <module>
    __all__ += [name for name in dir(_C)
NameError: name '_C' is not defined
2021-05-24 14:18:24,122	WARNING worker.py:1115 -- Failed to unpickle the remote function '__main__.remote_process_data_sample' with function ID 7e54bbdac7183014626c026a1a5f7a5d94bd047c2c579b3d7b5d492e. Traceback:
Traceback (most recent call last):
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/ray/_private/function_manager.py", line 180, in fetch_and_register_remote_function
    function = pickle.loads(serialized_function)
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/ray/cloudpickle/cloudpickle.py", line 584, in subimport
    __import__(name)
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/torch/__init__.py", line 635, in <module>
    import torch.utils.data
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/torch/utils/data/__init__.py", line 6, in <module>
    from .dataloader import DataLoader, _DatasetKind, get_worker_info
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/torch/utils/data/dataloader.py", line 21, in <module>
    from . import _utils
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/torch/utils/data/_utils/__init__.py", line 45, in <module>
    from . import worker, signal_handling, pin_memory, collate, fetch
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/torch/utils/data/_utils/worker.py", line 114, in <module>
    class _IterableDatasetStopIteration(object):
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/dataclasses.py", line 950, in wrap
    return _process_class(cls, init, repr, eq, order, unsafe_hash, frozen)
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/dataclasses.py", line 801, in _process_class
    for name, type in cls_annotations.items()]
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/dataclasses.py", line 801, in <listcomp>
    for name, type in cls_annotations.items()]
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/dataclasses.py", line 659, in _get_field
    if (_is_classvar(a_type, typing)
  File "/home/administrator/wav2vec_stt/lib/python3.7/site-packages/dataclasses.py", line 550, in _is_classvar
    return type(a_type) is typing._ClassVar
AttributeError: module 'typing' has no attribute '_ClassVar'

please Help me out
Thanks
@zilunpeng

**Comment by zilunpeng** [comment]
@vigneshgig Are you on a GPU machine? Ray inference worked for me on a CPU-only machine.

**Comment by vigneshgig** [comment]
Hi, sorry for the late reply, yeah I used GPU machine, actually in my case this error came for torch import. so I solved this problem by changing the import torch to import torch as to in main import. I think the ray instance tries to import the torch. but some there is a conflict between the ray instance of torch import and main import torch. so that's why I changed name of the main import.
Thanks for the reply
